### PR TITLE
Integrate Auth0 into AuthProfile

### PR DESCRIPTION
**Pull request by ajvb**
* Support a "filter in" list of auth0 client ids
* Move to using our own auth0.LogEvent so that it can implement
Serializable

### CODE REVIEW

File: src/main/java/com/mozilla/secops/parser/Auth0.java [hunk 1]
```diff
@@ -243,4 +243,22 @@ private ObjectMapper getObjectMapper() {
     _mapper.disable(SerializationFeature.FAIL_ON_EMPTY_BEANS);
     return _mapper;
   }
+
+  /**
+   * Return true if Auth0 event's client id is in the passed in list of client ids
+   *
+   * @param clientIds list of client ids to check for
+   * @return True if client id is in `clientIds`
```
Comment (ameihm0912) [inline src/main/java/com/mozilla/secops/parser/Auth0.java]: Does this backtick syntax work here? (cool if it does 👌)

File: src/main/java/com/mozilla/secops/parser/Auth0.java [hunk 2]
```diff
@@ -243,4 +243,22 @@ private ObjectMapper getObjectMapper() {
...
+   */
+  public Boolean hasClientIdIn(String[] clientIds) {
+    if (clientIds == null) {
+      return false;
+    }
+    for (String clientId : clientIds) {
+      if (event.getClientId().equals(clientId)) {
```
Comment (ameihm0912) [inline src/main/java/com/mozilla/secops/parser/Auth0.java]: Do we need a null check here?

File: pom.xml [hunk 3]
```diff
@@ -250,11 +250,6 @@
             <version>2.5.6</version>
             <type>pom</type>
         </dependency>
-        <dependency>
```
[the remark this answers is not in the document]
Comment (ameihm0912) [reply pom.xml]: @ajvb disregard this comment, I see now why you implemented it this way

File: src/main/java/com/mozilla/secops/parser/Auth0.java (hunk 2 above)
Comment (adrianosela) [reply src/main/java/com/mozilla/secops/parser/Auth0.java]: +1

File: src/main/java/com/mozilla/secops/authprofile/AuthProfile.java [hunk 4]
```diff
@@ -729,6 +740,11 @@ public void processElement(ProcessContext c) throws StateException {
     Integer getMaximumKilometersPerHour();
 
     void setMaximumKilometersPerHour(Integer value);
+
+    @Description("Auth0 Client ids to consider for state analysis (multiple allowed)")
+    String[] getAuth0ClientIds();
```
[the remark this answers is not in the document]
Comment (ajvb) [reply src/main/java/com/mozilla/secops/authprofile/AuthProfile.java]: No, I did that on purpose. Don't want to accidentally start processing all auth events.

File: src/main/java/com/mozilla/secops/parser/Auth0.java (hunk 1 above)
Comment (ajvb) [reply src/main/java/com/mozilla/secops/parser/Auth0.java]: No it doesn't lol, just a habit of mine. Removing them.

**Pull request merged**